test/tools/project-cleaner/main.go

```diff
@@ -112,6 +114,69 @@ func (p *ProjectCleaner) GKEClusterHandler(resource GCPResource) {
 	}
 }
 
+func (p *ProjectCleaner) PersistentDiskHandler(resource GCPResource) {
+	// See https://cloud.google.com/docs/authentication/.
+	// Use GOOGLE_APPLICATION_CREDENTIALS environment variable to specify a service account key file
+	// to authenticate to the API.
+	ctx := context.Background()
+	c, err := compute.NewDisksRESTClient(ctx)
+	if err != nil {
+		log.Fatalf("Could not initialize perisistent disk client: %v", err)
+	}
+	defer c.Close()
+
+	for _, zone := range resource.Zones {
+		// maxResults := uint32(2)
+		// order := "creationTimestamp desc"
+		req := &computepb.ListDisksRequest{
+			Project: p.ProjectId,
+			Zone:    zone,
+			// OrderBy:    &order,
+			// MaxResults: &maxResults,
+		}
+		disk_list, listerr := c.List(ctx, req)
+		if listerr != nil {
+			log.Fatalf("Could not get perisistent disk list: %v, zone: %s", listerr, zone)
+		}
+		diskmarshal, _ := json.Marshal(disk_list)
+		log.Printf("disk_list:%s", string(diskmarshal))
+		log.Printf("disk_list size:%d", len(disk_list.GetItems()))
+
+		for _, disk := range disk_list.GetItems() {
+
+			creationTimestamp := disk.GetCreationTimestamp()
+			createdTime, _ := time.Parse(time.RFC3339, creationTimestamp)
+			duration := time.Since(createdTime)
+
+			// If the disk is not detached yet, do not delete the persistent disk.
+			if len(disk.GetLastDetachTimestamp()) == 0 {
```

Comment:
Ideally, we could filter these out when listing disks. I think it should be supported, because we could do this with gcloud.

Reply:
Yes we should be able to configure filter for ListDisksRequest https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/compute/v1#ListDisksRequest. But it was not successful when I am trying to set something like `(lastDetachTimestamp != '') (users =  '')` along with orderby `creationTimestamp desc`. I got 400 bad request, but I don't know what exactly was the error in order to fix my request.

Reply:
Not sure, looking at documentation, they seem to use `""` for strings instead of `''`.
Did you try verifying your added configs? e.g. you can only add orderby, see if you get an error.
Only add one filter, see if you get an error, etc to identify what was wrong.

Reply:
https://pkg.go.dev/google.golang.org/genproto/googleapis/cloud/compute/v1#Disk
users seem to be string array, do you know filter syntax for arrays?

Reply:
> e.g. you can only add orderby, see if you get an error.
> Only add one filter, see if you get an error, etc to identify what was wrong.

If I only run `orderBy`, it is successful. If I only run `lastDetachTimestamp != ''` filter, it is also successful. But I can not have both `orderBy`and `filter` at the same time, which is strange.

Reply:
I think the GCloud CLI has richer filter syntax: https://cloud.google.com/sdk/gcloud/reference/topic/filters. I tried multiple cases and combination for checking array, but was not successful. And the combination of `filterBy` and `orderBy` is not a valid request either. Since we are going to compare the result later in the code, I am leaning towards comparing the result after getting the response, how do you think?

Reply:
Sure, we can skip the problem since you found a workaround : )